source/MaterialXRuntime/RtFileIo.cpp

```diff
@@ -590,6 +590,50 @@ namespace
         }
 
         writeAttributes(node, destNode);
+
+        return destNode;
+    }
+
+    void writeMaterialElements(const PvtNode* node, NodePtr mxNode, const string& nodeName, DocumentPtr doc, const RtWriteOptions* writeOptions)
+    {
+        MaterialPtr material = doc->addMaterial(mxNode->getName() + "_Material");
+        ShaderRefPtr shaderRef = material->addShaderRef("sref", nodeName);
+        for (InputPtr input : mxNode->getActiveInputs())
+        {
+            BindInputPtr bindInput = shaderRef->addBindInput(input->getName(), input->getType());
+            if (input->hasNodeName())
+            {
+                if (input->hasOutputString())
+                {
+                    bindInput->setNodeGraphString(input->getNodeName());
+                    bindInput->setOutputString(input->getOutputString());
+                }
+            }
+            else
+            {
+                bindInput->setValueString(input->getValueString());
+            }
+        }
+        for (ParameterPtr param : mxNode->getActiveParameters())
+        {
+            BindParamPtr bindParam = shaderRef->addBindParam(param->getName(), param->getType());
+            bindParam->setValueString(param->getValueString());
+        }
+        // Should we delete the surface shader?
+        if (writeOptions->materialWriteOp & RtWriteOptions::MaterialWriteOp::DELETE)
+        {
+            doc->removeChild(node->getName());
+        }
+        // Should we create a look for the material element?
+        if (writeOptions->materialWriteOp & RtWriteOptions::MaterialWriteOp::LOOK)
+        {
+            LookPtr look = doc->addLook();
+            MaterialAssignPtr materialAssign = look->addMaterialAssign();
+            materialAssign->setMaterial(material->getName());
+            CollectionPtr collection = doc->addCollection();
+            collection->setIncludeGeom("*");
```

Comment:
Minor: I think this should be "/*".

Reply:
Oh, ok, I didn't think that was the case, but I can change it.